website/server/models/group.js

```diff
@@ -323,6 +323,95 @@ schema.statics.getGroups = async function getGroups (options = {}) {
   return groupsArray;
 };
 
+function _translateMessage (lang, info) {
+  let msg;
+  let foundText = '';
+  let spells = shared.content.spells;
+  let quests = shared.content.quests;
+
+  switch (info.type) {
+    case 'quest_start':
+      msg = `\`${shared.i18n.t('chatQuestStarted', {questName: questScrolls[info.quest].text(lang)}, lang)}\``;
+      break;
+
+    case 'boss_damage':
+      msg = `\`${shared.i18n.t('chatBossDamage', {username: info.user, bossName: questScrolls[info.quest].boss.name(lang), userDamage: info.userDamage, bossDamage: info.bossDamage}, lang)}\``;
+      break;
+
+    case 'boss_dont_attack':
+      msg = `\`${shared.i18n.t('chatBossDontAttack', {username: info.user, bossName: questScrolls[info.quest].boss.name(lang), userDamage: info.userDamage}, lang)}\``;
+      break;
+
+    case 'boss_rage':
+      msg = `\`${questScrolls[info.quest].boss.rage.effect(lang)}\``;
+      break;
+
+    case 'boss_defeated':
+      msg = `\`${shared.i18n.t('chatBossDefeated', {bossName: questScrolls[info.quest].boss.name(lang)}, lang)}\``;
+      break;
+
+    case 'user_found_items':
+      foundText = _.reduce(info.items, (m, v, k) => {
+        m.push(`${v} ${questScrolls[info.quest].collect[k].text(lang)}`);
+        return m;
+      }, []).join(', ');
+      msg = `\`${shared.i18n.t('chatFindItems', {username: info.user, items: foundText}, lang)}\``;
+      break;
+
+    case 'all_items_found':
+      msg = `\`${shared.i18n.t('chatItemQuestFinish', lang)}\``;
+      break;
+
+    case 'spell_cast_party':
+      msg = `\`${shared.i18n.t('chatCastSpellParty', {username: info.user, spell: spells[info.class][info.spell].text(lang)}, lang)}\``;
+      break;
+
+    case 'spell_cast_user':
+      msg = `\`${shared.i18n.t('chatCastSpellUser', {username: info.user, spell: spells[info.class][info.spell].text(lang), target: info.target}, lang)}\``;
+      break;
+
+    case 'quest_abort':
+      msg = `\`${shared.i18n.t('chatQuestAborted', {username: info.user, questName: questScrolls[info.quest].text(lang)}, lang)}\``;
+      break;
+
+    case 'tavern_quest_completed':
+      msg = `\`${quests[info.quest].completionChat(lang)}\``;
+      break;
+
+    case 'tavern_boss_rage_tired':
+      msg = `\`${shared.i18n.t('tavernBossTired', {rageName: quests[info.quest].boss.rage.title(lang), bossName: quests[info.quest].boss.name(lang)}, lang)}\``;
+      break;
+
+    case 'tavern_boss_rage':
+      msg = `\`${quests[info.quest].boss.rage[info.scene](lang)}\``;
+      break;
+
+    case 'tavern_boss_desperation':
+      msg = `\`${quests[info.quest].boss.desperation.text(lang)}\``;
+      break;
+
+    case 'claim_task':
+      msg = `${shared.i18n.t('userIsClamingTask', {username: info.user, task: info.task}, lang)}`;
+      break;
+  }
+
+  return msg;
+}
+
+function _translateSystemMessages (group, user) {
+  let lang = user.preferences.language;
```

Comment:
After almost 2 hours fixing conflicts, just noticed that the `user` sent to `Group.toJSONCleanChat` as parameter don't have the `language` information anymore. Without this information I can't do any translation.
May I do a query to the DB to take the user's language?

P.S.: More 4 months and will be my PR's birthday, 1 year old!! hahah

Reply:
@Yutsuten I'm so sorry about this but we've not yet had the time to make the changes to the chat routes necessary for this PR. Don't worry about fixing the conflicts, when we have a chance and the other issues are fixed we'll fix them Sorry again

Reply:
Got it, thank you for the reply! I think I'll push the conflict fixes anyway, some of them were tricky to solve, and it may help you guys when merging. The only problem now is that the `lang` variable I mentioned receives `undefined`.

Reply:
Is this still an issue? Can you point to me where it's happening?

Reply:
It was a problem on Oct/2018, but don't know how it is now. I'll have to build Habitica locally again to check. If it is still an issue, I'll write here.

Reply:
Just confirmed that the problem _still_ exists with the latest commit of this PR.

After editing the above part of the source code to
```javascript
function _translateSystemMessages (group, user) {
  console.log('####################');
  console.log(user);
  let lang = user.preferences.language;
  console.log(lang);
  console.log('####################');
```

I've got this output:

```
server_1  | ####################
server_1  | { contributor: {},
server_1  |   party:
server_1  |    { quest:
server_1  |       { progress: [Object],
server_1  |         RSVPNeeded: false,
server_1  |         key: 'basilist',
server_1  |         completed: null },
server_1  |      order: 'level',
server_1  |      orderAscending: 'ascending',
server_1  |      _id: '066ceec3-9eb8-45e7-852b-a07a9202f6a8' },
server_1  |   guilds: [],
server_1  |   _id: 'ef191782-0b23-4240-a2b4-aacb0ba0d99d' }
server_1  | undefined
server_1  | ####################
```

Any recommended way to access the user's language in this case?